cmd/libsnap-confine-private/utils.h

```diff
@@ -149,4 +152,21 @@ __attribute__((warn_unused_result)) bool sc_is_expected_path(const char *path);
  */
 bool sc_wait_for_file(const char *path, size_t timeout_sec);
 
+/**
+ * Ensure a directory exists inside a given parent directory. Essentially a
+ * convenience wrapper around mkdirat() followed by fchownat(), if a new
+ * directory was created.
+ *
+ * Returns -1 in case of error.
+ */
+int sc_ensure_mkdirat(int fd, const char *name, mode_t mode, uid_t uid, uid_t gid);
+
+/**
+ * Ensure a directory exists. Essentially a convenience wrapper around mkdirat()
+ * followed by chown() if a new directory was created.
+ *
+ * Returns -1 in case of error.
+ */
+int sc_ensure_mkdir(const char *name, mode_t mode, uid_t uid, uid_t gid);
```

Comment:
```suggestion
__attribute__((warn_unused_result)) int sc_ensure_mkdir(const char *name, mode_t mode, uid_t uid, uid_t gid);
```

Reply:
applied in https://github.com/canonical/snapd/pull/15187